### PR TITLE
refactor(core): Clean up `getNgZone` implementation.

### PR DESCRIPTION
**Pull request by dylhunn**
The current implementation is a bit confusing to read. Get rid of some of the operator soup.

### CODE REVIEW

**Review by atscott — APPROVED** [review]

File: packages/core/src/application_ref.ts [hunk 1]
```diff
@@ -583,15 +583,15 @@ function getNgZoneOptions(options?: BootstrapOptions): NgZoneOptions {
...
+function getNgZone(
+    ngZoneToUse: NgZone|'zone.js'|'noop' = 'zone.js', options: NgZoneOptions): NgZone {
   if (ngZoneToUse === 'noop') {
-    ngZone = new NoopNgZone();
+    return new NoopNgZone();
+  } else if (ngZoneToUse === 'zone.js') {
+    return new NgZone(options);
   } else {
-    ngZone = (ngZoneToUse === 'zone.js' ? undefined : ngZoneToUse) || new NgZone(options);
+    return ngZoneToUse;
```
Comment (AndrewKushnir) [inline packages/core/src/application_ref.ts:593]: nit: you probably don't need the last `else` (can just `return ngZoneToUse;`).

File: packages/core/src/application_ref.ts [hunk 2]
```diff
@@ -583,15 +583,15 @@ function getNgZoneOptions(options?: BootstrapOptions): NgZoneOptions {
...
 
-function getNgZone(ngZoneToUse: NgZone|'zone.js'|'noop'|undefined, options: NgZoneOptions): NgZone {
-  let ngZone: NgZone;
-
+function getNgZone(
+    ngZoneToUse: NgZone|'zone.js'|'noop' = 'zone.js', options: NgZoneOptions): NgZone {
   if (ngZoneToUse === 'noop') {
-    ngZone = new NoopNgZone();
+    return new NoopNgZone();
+  } else if (ngZoneToUse === 'zone.js') {
```
Comment (alan-agius4) [inline packages/core/src/application_ref.ts:590]: nit: technically you don't need `else if` but just `if` due to the previous `return`

**Review by AndrewKushnir — APPROVED** [review]
👍

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 675fe2fabcf56a7177a48a9032ef2bb406dcb7f8.

**Pull request closed without merging**